### PR TITLE
🐛(frontend) show message when a product without any course run

### PR DESCRIPTION
**Pull request by PanchoutNathan**
## Purpose
A user is able today to buy a product with no open course runs. 

## Proposal

- [x] hide the purchase button
- [x] show message to explain why

### CODE REVIEW

File: src/frontend/js/components/CourseProductItem/_styles.scss [hunk 1]
```diff
@@ -95,6 +95,11 @@
   &__footer {
```
Comment (jbpenrath) [inline src/frontend/js/components/CourseProductItem/_styles.scss:95]: Add into the commit footer
`Close #1859`

File: src/frontend/js/components/SaleTunnel/index.tsx [hunk 2]
```diff
@@ -117,6 +132,14 @@ const SaleTunnel = ({ product, courseCode }: Props) => {
     );
   }
 
+  if (!hasCourseRun) {
+    return (
+      <div className="product-item__no-course-run">
```
Comment (jbpenrath) [inline src/frontend/js/components/SaleTunnel/index.tsx:137]: Use p tag.

File: src/frontend/js/components/SaleTunnel/index.tsx [hunk 3]
```diff
@@ -86,6 +92,15 @@ const SaleTunnel = ({ product, courseCode }: Props) => {
   const productQuery = useProduct(product.id);
   const ordersQuery = useOrders({ product: product.id, course: courseCode });
   const [isOpen, setIsOpen] = useState(false);
+
+  const hasCourseRun = useMemo(() => {
+    let count = 0;
+    product.target_courses.forEach((course) => {
+      count += course.course_runs.length;
+    });
+    return count > 0;
```
Comment (jbpenrath) [inline src/frontend/js/components/SaleTunnel/index.tsx:101]: Block the sell when one target course of the product has no course runs.
```suggestion
return product.target_courses.some(({ course_runs }) => course_runs.length === 0);
```

Comment (PanchoutNathan) [reply src/frontend/js/components/SaleTunnel/index.tsx:101]: Done

**Review by jbpenrath — APPROVED** [review]
LGTM ✅

**Pull request merged**